platform/darwin/src/MGLShapeSource.mm

```diff
@@ -56,6 +56,21 @@ - (instancetype)initWithIdentifier:(NSString *)identifier shape:(nullable MGLSha
     return self;
 }
 
+- (instancetype)initWithIdentifier:(NSString *)identifier features:(NS_ARRAY_OF(id <MGLFeature>) *)features options:(nullable NS_DICTIONARY_OF(MGLShapeSourceOption, id) *)options {
+    for (id <MGLFeature> feature in features) {
+        if (![feature conformsToProtocol:@protocol(MGLFeature)]) {
+            [NSException raise:NSInvalidArgumentException format:@"The object %@ included in the features argument does not conform to the MGLFeature protocol.", feature];
+        }
+    }
+    MGLShapeCollectionFeature *shapeCollectionFeature = [MGLShapeCollectionFeature shapeCollectionWithShapes:(NSArray *)features];
```

Comment:
The implementation of `+[MGLShapeCollectionFeature shapeCollectionWithShapes:]` uses an unqualified `NSArray` in its signature, but the declaration has a qualified `NSArray<MGLShape<MGLFeature> *> *`. Omitting the qualifier in the implementation only takes away opportunities for the compiler to catch mistakes. We should qualify as much as possible.

Since `+[MGLShapeCollectionFeature shapeCollectionWithShapes:]` is declared to take an `NSArray<MGLShape<MGLFeature> *> *` rather than the broader type `NSArray<id<MGLFeature>> *`, `-[MGLShapeSource initWithIdentifier:features:options:]` should take an `NSArray<MGLShape<MGLFeature> *> *` as well.